tmt/beakerlib.py

```diff
@@ -342,7 +342,7 @@ def already_fetched(lib: str) -> bool:
         except LibraryError:
             if dependency in original_require:
                 processed_require.add(dependency)
-            if dependency in original_recommend:
+            if isinstance(dependency, str) and dependency in original_recommend:
```

Comment:
`dependency` is of type `Require`, which is a `Union[str, RequireFmfId]`. This is perfectly compatible with `require`-like containers, but not with `recommend`-like ones because fmf id is not allowed under `recommend` key. Since the content of `require` and `recommend` keys is squashed into a single iterable, a check is needed to make sure we're comparing only the `str` subtype against `require` content.

Reply:
Thanks for clarification.